src/catalogue/catalogue.component.tsx

```diff
@@ -63,6 +63,21 @@ const AddCategoryButton = (props: AddCatalogueButtonProps) => {
   );
 };
 
+function generateUniqueName(
+  existingNames: (string | undefined)[],
+  originalName: string
+) {
+  let newName = originalName;
+  let copyIndex = 1;
+
+  while (existingNames.includes(newName)) {
+    newName = `${originalName}_copy_${copyIndex}`;
+    copyIndex++;
+  }
+
+  return newName;
+}
```

Comment:
I just added this function to `utils.tsx` in #137, could you merge develop and use that instead of repeating it here please.